llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -670,7 +670,7 @@ static bool tryToShorten(Instruction *DeadI, int64_t &DeadStart,
   assert(DeadSize > ToRemoveSize && "Can't remove more than original size");
 
   uint64_t NewSize = DeadSize - ToRemoveSize;
-  if (auto *AMI = dyn_cast<AtomicMemIntrinsic>(DeadI)) {
+  if (auto *AMI = dyn_cast<AnyMemIntrinsic>(DeadI); AMI && AMI->isAtomic()) {
```

Comment:
DeadI is already AnyMemIntrinsic.

Reply:
It's not actually, but there's a DeadIntrinsic cast which is.  Adjusting.